archiver.go

```diff
@@ -292,12 +295,21 @@ func (a *Archiver) pushChangesToRootedRepository(ctx context.Context, r *model.R
 		}
 
 		refspecs := a.changesToPushRefSpec(r.ID, changes)
+		pushStart := time.Now()
 		if err := tr.Push(ctx, url, refspecs); err != nil {
+			onlyPushDurationSec := int64(time.Now().Sub(pushStart) / time.Second)
+			log.Error("error pushing 1 change for", "refs", refspecs, "error", err, "took", onlyPushDurationSec)
 			_ = tx.Rollback()
 			return err
 		}
+		onlyPushDurationSec := int64(time.Now().Sub(pushStart) / time.Second)
```

Comment:
is there any specific reason for printing this as number of seconds instead of regular duration? (e.g. 3m45s)

Reply:
The only reason is - in my experiment `time.Duration` was printed in different Units, depending on how long it is. I'm not super-familiar with go stdlib and might have missed something, but made it trickier to parse: one line was `sec` another `μsec`.

Reply:
If you only need seconds you can get log time, right? Anyways, we should think on do all this stuff using a library like https://github.com/rcrowley/go-metrics

Reply:
Important part is - we need a duration here, not just the timestamp. Otherwise it quickly becomes complex on analysis part (see [here](https://gist.github.com/bzz/a1add99b24370f8dcbe0976f19b4611f) for example).

I 100% agree on metrics, let's try to push it in the next sprint's scope and I'll be happy to refactor this in subsequent PRs.